src/netaddress.cpp

```diff
@@ -101,24 +193,76 @@ bool CNetAddr::SetInternal(const std::string &name)
     return true;
 }
 
+namespace torv3 {
+// https://gitweb.torproject.org/torspec.git/tree/rend-spec-v3.txt#n2135
+static constexpr size_t CHECKSUM_LEN = 2;
+static constexpr size_t VERSION_LEN = 1;
+static const unsigned char VERSION[] = "\x03";
+static constexpr size_t TOTAL_LEN = ADDR_TORV3_SIZE + CHECKSUM_LEN + VERSION_LEN;
+
+static void Checksum(Span<const uint8_t> addr_pubkey, uint8_t (&checksum)[CHECKSUM_LEN])
+{
+    // CHECKSUM = H(".onion checksum" | PUBKEY | VERSION)[:2]
```

Comment:
1046e17c nit suggestion if you retouch: s/CHECKSUM/tor v3 checksum/

Reply:
I copied that verbatim from https://gitweb.torproject.org/torspec.git/tree/rend-spec-v3.txt#n2135, but anyway - I now prefixed it with "TORv3".